pkg/actors/actors_test.go

```diff
@@ -282,6 +290,88 @@ func TestOverrideReminder(t *testing.T) {
 	})
 }
 
+func TestOverrideReminderCancelsActiveReminders(t *testing.T) {
+	ctx := context.Background()
+	t.Run("override data", func(t *testing.T) {
+		mockAppChannel := new(channelt.MockAppChannel)
+		testActorsRuntime := newTestActorsRuntimeWithMock(mockAppChannel)
+		actorType, actorID := getTestActorTypeAndID()
+		reminderName := "reminder1"
+
+		reminder := createReminderData(actorID, actorType, reminderName, "10s", "1s", "a")
+		err := testActorsRuntime.CreateReminder(ctx, &reminder)
+		assert.Nil(t, err)
+
+		reminder2 := createReminderData(actorID, actorType, reminderName, "9s", "1s", "b")
+		testActorsRuntime.CreateReminder(ctx, &reminder2)
+		reminders, err := testActorsRuntime.getRemindersForActorType(actorType)
+		assert.Nil(t, err)
+		// Check reminder is updated
+		assert.Equal(t, "9s", reminders[0].Period)
+		assert.Equal(t, "1s", reminders[0].DueTime)
+		assert.Equal(t, "b", reminders[0].Data)
+
+		reminder3 := createReminderData(actorID, actorType, reminderName, "8s", "2s", "b")
+		testActorsRuntime.CreateReminder(ctx, &reminder3)
+		reminders, err = testActorsRuntime.getRemindersForActorType(actorType)
+		assert.Nil(t, err)
+		// Check reminder is updated
+		assert.Equal(t, "8s", reminders[0].Period)
+		assert.Equal(t, "2s", reminders[0].DueTime)
+		assert.Equal(t, "b", reminders[0].Data)
+
+		time.Sleep(2 * time.Second)
+
+		// Test only the last reminder update fires
+		mockAppChannel.AssertNumberOfCalls(t, "InvokeMethod", 1)
+	})
+}
+
+func TestOverrideReminderCancelsMultipleActiveReminders(t *testing.T) {
+	ctx := context.Background()
+	t.Run("override data", func(t *testing.T) {
+		mockAppChannel := new(channelt.MockAppChannel)
+		testActorsRuntime := newTestActorsRuntimeWithMock(mockAppChannel)
+		actorType, actorID := getTestActorTypeAndID()
+		reminderName := "reminder1"
+
+		reminder := createReminderData(actorID, actorType, reminderName, "10s", "2s", "a")
+		err := testActorsRuntime.CreateReminder(ctx, &reminder)
+		assert.Nil(t, err)
+
+		time.Sleep(50 * time.Millisecond)
+
+		reminder2 := createReminderData(actorID, actorType, reminderName, "9s", "1s", "b")
+		reminder3 := createReminderData(actorID, actorType, reminderName, "8s", "1s", "b")
+		testActorsRuntime.CreateReminder(ctx, &reminder2)
+		testActorsRuntime.CreateReminder(ctx, &reminder3)
```

Comment:
reminder 2 and reminder 3 will not run in parallel. we need to use go routine here.
for example.

go func() {
testActorsRuntime.CreateReminder(ctx, &reminder2)
}
go func() {
testActorsRuntime.CreateReminder(ctx, &reminder3)
}

time.Sleep(xxx)

Reply:
And pr build is broken. please fix it.